File: Frontend/src/types/lessor.ts
import type { LeaseStatus } from "@/types/lease"
export type LessorLease = {
    leaseId: number
    leaseNo: string | null
    leasePropertyAddress: string | null
    startDate: string | null
    endDate: string | null
    agreementValue: number | string | null
    annualRate: number | string | null
    leaseStatus: LeaseStatus | null
}

export type Lessor = {
    lessorId: number
    fullName: string | null
    nic: string | null
    address: string | null
    bankName: string | null
    accountNumber: string | null
    bankCode: string | null
    createdAt: string
    updatedAt: string
    leases?: LessorLease[]
}

export type CreateLessorInput = {
    fullName: string
    nic: string
    address: string
    bankName: string
    accountNumber: string
    bankCode: string
}

export type UpdateLessorInput = CreateLessorInput & {
    lessorId: number
}

File: Backend/IFRS/models/Lessor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace IFRS.models;

[Table("lessor")]
public class Lessor
{
    [Key]
    [Column("pk_lessor_id")]
    public int LessorId { get; set; }

    [Column("full_name")]
    [StringLength(255)]
    public string? FullName { get; set; }

    [Column("nic")]
    [StringLength(20)]
    public string? Nic { get; set; }

    [Column("address")]
    public string? Address { get; set; }

    [Column("bank_name")]
    [StringLength(255)]
    public string? BankName { get; set; }

    [Column("account_number")]
    [StringLength(100)]
    public string? AccountNumber { get; set; }

    [Column("bank_code")]
    [StringLength(100)]
    public string? BankCode { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation property
    public ICollection<Lease> Leases { get; set; } = new List<Lease>();
}
